lib/kv-extreme-scale.js

```diff
@@ -275,61 +268,46 @@ function(modelName, filter, options) {
 
   function takeNextFromCache(callback) {
     const value = cache.shift();
-
-    ModelKeyComposer.parse(value, (err, parsed) => {
-      if (err && err.code == 'NO_MODEL_PREFIX') {
-        err = null;
-        parsed = {modelName: null, key: value};
-      }
-
-      if (err) return cb(err);
-
-      if (parsed.modelName !== modelName) {
-        g.warn(
-          'Warning: key scan returned a key beloging to a wrong model.' +
-            '\nExpected model name: %j' +
-            '\nActual model name:   %j' +
-            '\nThe key: %j',
-          modelName, parsed.modelName, value);
-      }
-
-      callback(null, parsed.key);
-    });
+    callback(null, value);
   }
 
   function reportEnd(callback) {
     setImmediate(callback);
   }
 
   function fetchFirst500Keys(callback) {
-    ModelKeyComposer.compose(modelName, KEY_QUERY, (err, pattern) => {
+    self._query(modelName, KEY_QUERY, (err, body) => {
       if (err) return callback(err);
-      self._query(pattern, (err, body) => {
-        if (err) return callback(err);
-        cache = body.items.map(it => it.key);
-        getNextKey(callback);
-      });
+      cache = body.items.map(it => it.key);
+      getNextKey(callback);
     });
   }
 };
 
-ExtremeScaleKVConnector.prototype._query = function(query, callback) {
+ExtremeScaleKVConnector.prototype._query =
+function(modelName, query, callback) {
+  const mapname = this.mapname(modelName);
```

Comment:
I believe we agreed to modify `_query` so that it does not need to call `mapname` and let `this.request` handle that part?

```js
requestOptions = {
  baseUrl: this._queryUrl,
  // ...
};
```